pkg/controllers/management/rkeworkerupgrader/node.go

```diff
@@ -161,6 +163,10 @@ func (uh *upgradeHandler) filterNodes(nodes []*v3.Node, expectedVersion int, dra
 			} else {
 				// node hasn't un-cordoned, so consider it upgrading in terms of maxUnavailable count
 				status.upgrading++
+				// node has already upgraded, but condition is not unknown, so uncordon it
+				if !v32.NodeConditionUpgraded.IsUnknown(node) && node.Spec.DesiredNodeUnschedulable != "false" {
```

Comment:
Can `NodeConditionUpgraded` be false? If so would we still want to attempt to un-cordon?

Reply:
It can't be false, this is because we wanted to be consistent with the old approach of how there's no errors when node plan isn't sent or applied correctly -> we just continue trying, and node would stay in upgrading state forever.